src/codeu/chat/util/MorseConverter.java

```diff
@@ -0,0 +1,217 @@
+    for (int i = 0; i < characters.length; i++) {
+      strBuild.append(CODE.get(characters[i]));
+      strBuild.append(" ");
+    }
+    strBuild.append("/");
+    return strBuild.toString();
+  }
+
+  /**
+   * Converts a paragraph into morse code. Delineates all of the punctuation.
+   *
+   * @param paragraph
+   *          String
+   * @return String of morse
+   */
+  public static String paragraphToMorse(String paragraph) {
+    String[] words = paragraph.split("\\W+");
+    StringBuilder strBuild = new StringBuilder();
+    for (String word : words) {
+      strBuild.append(wordToMorse(word));
+    }
+    return strBuild.toString();
+  }
+
+  /**
+   * Return the letter associated with the given morse.
+   *
+   * @param morse
+   *          String
+   * @return String of english letter
+   */
+  public static char morseToLetter(String morse) {
+    if (!morseCheck(morse)) {
+      throw new IllegalArgumentException(
+          "ERROR: string of morse must be only . and –");
+    }
+    Iterator<Entry<Character, String>> itr = CODE.entrySet().iterator();
+    while (itr.hasNext()) {
+      Entry<Character, String> entry = itr.next();
+      if (entry.getValue().equals(morse)) {
+        return entry.getKey();
+      }
+    }
+    return '\0';
+  }
+
+  private static boolean morseCheck(String morse) {
+    char[] check = morse.toCharArray();
+    for (int i = 0; i < check.length; i++) {
+      char character = check[i];
+      if (character != '.' || character != '–' 
+        || character != ' ' || character != '/') {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Converts a string of morse into a string of integers that represent the
+   * sounds that will be used to translate this. 0 is for dot. 1 is for dash. 2
+   * is for space in between. 3 is for word ending.
+   *
+   * @param morse
+   *          String
+   * @return List of integer
+   */
+  public static List<Integer> morseToSoundTranslate(String morse) {
+    String morseInput = morse;
+    //Check that the morse string is in the expected format
+    assert(MorseConverter.morseCheck(morse));
+
+    char[] morseLetters = morse.toCharArray();
+    List<Integer> soundTranslate = new ArrayList<>();
+    for (char c : morseLetters) {
+      switch (c) {
+      case '.':
+        soundTranslate.add(0);
+        break;
+      case '–':
+        soundTranslate.add(1);
+        break;
+      case ' ':
+        soundTranslate.add(2);
+        break;
+      case '/':
+        soundTranslate.add(3);
+      default:
+        break;
+      }
+    }
+    return soundTranslate;
+  }
+
+  /**
+   * The input is assumed to be only "." and/or "–" and "/" and " ". It will
+   * translate the input morse string into sound, and play it if the audio is
+   * available.
+   *
+   * @param morse
+   *          String
+   */
+  public static void morseToSound(String morse) {
+    byte[] buf = new byte[1];
+    AudioFormat af = new AudioFormat(44100, 8, 1, true, false);
```

Comment:
these argument values look a bit magical. this would be a good use case for class constants.